app/controllers/albums.js

```diff
@@ -11,5 +13,26 @@ module.exports = {
         })
       });
     });
+  },
+
+  buy(req, res, next) {
+    const purchase = {
+      userId: req.user.id,
+      albumId: req.params.id
+    };
+    Purchase.findOne({ where: purchase })
+      .then(p => {
+        if (p) {
+          next(albumAlreadyPurchased);
```

Comment:
you can use `return next(album...` and avoid the else statement

Reply:
fixed